drivers/soundwire/cadence_master.c

```diff
@@ -1325,20 +1325,38 @@ static int cdns_port_params(struct sdw_bus *bus,
 			    struct sdw_port_params *p_params, unsigned int bank)
 {
 	struct sdw_cdns *cdns = bus_to_cdns(bus);
-	int dpn_config = 0, dpn_config_off;
+	int dpn_config;
+	int dpn_config_off_source;
+	int dpn_config_off_target;
+	int target_num = p_params->num;
+	int source_num = p_params->num;
+	bool override = false;
+
+	if (target_num == cdns->pdi_loopback_target &&
+	    !cdns->pdi_loopback_source) {
```

Comment:
`!cdns->pdi_loopback_source` should be `cdns->pdi_loopback_source`

Reply:
same